### PR TITLE
Remove mdbook, transition docs to Docusaurus

### PR DESCRIPTION
**Pull request by rmshea**
#### Problem

Gitbook, while it remains a formidable documentation rendering solution, has provided a myriad of issue for our purposes. We're moving back to Solana-hosted and controlled docs with a formal transition to [Docusaurus](https://v2.docusaurus.io/), a React-based Markdown renderer with plenty of flexibility and functionality.

You can see the current Docusaurus implementation [here.](https://beta.docs.solana.com)

#### Summary of Changes

- Remove mdbook scripts and other infrastructure
- Add Docusaurus implementation
- Remove Gitbook-specific syntax

### CODE REVIEW

**Comment by rmshea** [comment]
@danpaul000 looks like we'll need to adjust the `/docs` build subsequence. Could use your help to wrap that up—let's sync up on Monday.

**Comment by mvines** [comment]
Docusaurus has no search feature?  That's a bummer.   I use that in gitbook all the time :-/

**Comment by danpaul000** [comment]
Why did everything in `docs/` move to `docs/docs/`?

**Comment by mvines** [comment]
How does the deployment work?

**Comment by rmshea** [comment]
@mvines Docusaurus does have search — we just have to deploy it first and then we can use https://docsearch.algolia.com/apply/

**Comment by rmshea** [comment]
@danpaul000 That's how Docusaurus structures their markdown compartmentalization. I can rename the internal folder to whatever you might want, e.g. `src`

**Comment by rmshea** [comment]
> 
> 
> How does the deployment work?

We can either use GitHub Pages, Netlify, Vercel, or any other static site host. I've used Vercel for the beta.docs.solana.com deployment just for ease of use.

https://v2.docusaurus.io/docs/deployment

**Comment by danpaul000** [comment]
> @danpaul000 That's how Docusaurus structures their markdown compartmentalization. I can rename the internal folder to whatever you might want, e.g. `src`

Yeah, `docs/src/` is preferable.  That's where the files currently exist, and would massively reduce the number of touched files in this PR making it easier to focus on the deployment change.

**Comment by mvines** [comment]
Oh we're going to want to backport this to 1.2, heads up!

**Comment by danpaul000** [comment]
> Oh we're going to want to backport this to 1.2, heads up!

Can we wait on the backport to soak test master and whatever deployment changes?  I really don't want to break the main docs branch, we are getting a lot of traffic there lately.

**Comment by mvines** [comment]
> > Oh we're going to want to backport this to 1.2, heads up!
> 
> Can we wait on the backport to soak test master and whatever deployment changes? I really don't want to break the main docs branch, we are getting a lot of traffic there lately.

💯

**Comment by garious** [comment]
@rmshea, Rebase?

**Comment by garious** [comment]
@danpaul000, are you working on CI changes for this?

**Comment by danpaul000** [comment]
> @danpaul000, are you working on CI changes for this?

yeah, looking in to it

File: docs/src/mdx.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (danpaul000) [inline docs/src/mdx.md:1]: Don't think this file is needed.

File: docs/src/pages/index.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    imageUrl: "docs/cluster/README",
+    description: (
+      <>
+        Familiarize yourself with the high level architecture of a Solana
+        cluster.
+      </>
+    ),
+  }, //
+  // {
+  //   title: <>Understand Our Economic Design</>,
```
Comment (danpaul000) [inline docs/src/pages/index.js:61]: Why is this one commented out?

**Comment by danpaul000** [comment]
Open items for integration:
 - reinstate `docs/build.sh` for CI with `yarn start` and/or `yarn build` and ideally some kind of link checker like we had for mdbook.
 - Make sure `set_solana_release_tag.sh` and `build-cli-usage.sh` are called
 - For post-merge CI build, where should static files end up living?  Currently `yarn build` puts them in the `.gitignore`'d `docs/build` directory.
 - Need to hook up a publishing tool in CI with perms to write to the solana.com DNS.  Currently, @rmshea has done this manually using his personal permissions and the `vercel` tool.  Can this be integrated to be called from buildkite, or a different publishing tool?

I'm not sure what is the right approach to use the native docusaurus versioning setup, which has all versions as separate subdirs, rather than separate branches (https://v2.docusaurus.io/docs/versioning#directory-structure), unless we build this dynamically and push to a docs repo.  Their UI for jumping between versions of docs is nice, and doesn't require separate domains, ie beta.docs.solana.com, edge.docs.solana.com.  It should look similar to gitbook setup, ie docs.solana.com/v1.2.0, docs.solana.com/v.1.1.0.

@garious I can work on some of this later but don't think I'm going to be able to get it all over the line today.  More likely EOW.

File: docs/src/pages/index.js (hunk 2 above)
Comment (rmshea) [reply docs/src/pages/index.js:61]: To keep visual consistency — it'd mess up the 3x2 table layout.

File: docs/src/css/custom.css [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  --ifm-color-primary: #25c2a0;
+  --ifm-color-primary-dark: #409088;
+  --ifm-color-primary-darker: #387462;
+  --ifm-color-primary-darkest: #1b4e3f;
+  --ifm-color-primary-light: #42ba96;
+  --ifm-color-primary-lighter: #86b8b6;
+  --ifm-color-primary-lightest: #abd5c6;
+  --ifm-code-font-size: 95%;
+  --ifm-spacing-horizontal: 1em;
+  --ifm-font-family-base: "Rubik", system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif, BlinkMacSystemFont, 'Segoe UI', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
```
Comment (CriesofCarrots) [inline docs/src/css/custom.css:22]: Does our style guide offer any different typefaces for multi-paragraph text? Rubik looks great in small chunks on solana.com, but I find it uncomfortable to read in the context of the docs, especially in dark mode.

**Comment by rmshea** [comment]
@danpaul000 — let me know what you'd like my help with in terms of addressing your items above.

File: docs/src/css/custom.css (hunk 3 above)
Comment (rmshea) [reply docs/src/css/custom.css:22]: @CriesofCarrots let me ask Raj and see what we can do. I agree with this completely. Maybe we can keep Roboto.

**Comment by garious** [comment]
@danpaul000, any news here? Is this being worked on someplace else?

File: docs/src/css/custom.css (hunk 3 above)
Comment (rmshea) [reply docs/src/css/custom.css]: @CriesofCarrots let me know if you are a fan of Roboto!

Comment (CriesofCarrots) [reply docs/src/css/custom.css]: Yes, I am!

**Pull request closed without merging**